langchain/vectorstores/qdrant.py

```diff
@@ -461,18 +476,40 @@ def _document_from_scored_point(
             metadata=scored_point.payload.get(metadata_payload_key) or {},
         )
 
-    def _qdrant_filter_from_dict(self, filter: Optional[MetadataFilter]) -> Any:
-        if filter is None or 0 == len(filter):
-            return None
-
+    def _build_condition(self, key: str, value: Any) -> List[rest.FieldCondition]:
         from qdrant_client.http import models as rest
 
-        return rest.Filter(
-            must=[
+        out = []
+
+        if isinstance(value, dict):
+            for _key, value in value.items():
+                out.extend(self._build_condition(f"{key}.{_key}", value))
+        elif isinstance(value, list):
+            for _value in value:
+                if isinstance(_value, dict):
+                    out.extend(self._build_condition(f"{key}[]", _value))
+                else:
+                    out.extend(self._build_condition(f"{key}", _value))
+        else:
+            out.append(
                 rest.FieldCondition(
                     key=f"{self.metadata_payload_key}.{key}",
                     match=rest.MatchValue(value=value),
                 )
+            )
+
+        return out
+
+    def _qdrant_filter_from_dict(self, filter: Optional[MetadataFilter]) -> rest.Filter:
+        if not filter:
+            return None
+
+        from qdrant_client.http import models as rest
```

Comment:
small thing, could we keep imports at beginning of function (line 504)

Reply:
@dev2049, updated now. :)